### PR TITLE
Feature/#188/add skills endpoint

### PR DESCRIPTION
**Pull request by JustTheDude001**
I added the endpoint in order to update the skills, otherwise called tags, from a given resume given a student id.

git diff --stat main:
```
 app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php |  76 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 app/Http/Controllers/api/Student/StudentSkillsController.php                  |  40 ++++++++++++++++++++++++++++++++++++++++
 app/Http/Requests/SkillsRequest.php                                           |  48 ++++++++++++++++++++++++++++++++++++++++++++++++
 app/Service/Student/StudentSkillsService.php                                  |  46 ++++++++++++++++++++++++++++++++++++++++++++++
 routes/api/v1.php                                                             |   4 +++-
 tests/Feature/Controller/Student/StudentSkillsControllerTest.php              | 109 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 tests/Feature/Service/Student/StudentSkillsServiceTest.php                    |  98 ++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
```

- Tests:
image

- Swagger:
image

### CODE REVIEW

File: .dockerignore
Comment (jordimorillo) [inline .dockerignore:1]: Restore the permission change

File: app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
+<?php
+
```
Comment (jordimorillo) [inline app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php:2]:
```php
declare(strict_types=1);
```

File: app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+     *         @OA\Schema(
+     *             type="string",
+     *             format="uuid"
+     *         )
+     *     ),
+     * 
+     *     @OA\RequestBody(
+     *           @OA\JsonContent(
+     *              type="object",
+     *              @OA\Property(property="skills", type="array",
```
Comment (jordimorillo) [inline app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php:30]: Fix the tabs

File: app/Http/Controllers/api/Student/StudentSkillsController.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+use Illuminate\Http\JsonResponse;
+
+use App\Http\Requests\SkillsRequest;
+
+
+class StudentSkillsController extends Controller
+{
+    private StudentSkillsService $studentSkillsService;
+
+    public function __construct(StudentSkillsService $studentSkillsService)
```
Comment (jordimorillo) [inline app/Http/Controllers/api/Student/StudentSkillsController.php:21]: UpdateStudentSkillsService

File: app/Http/Requests/SkillsRequest.php [hunk 4]
```diff
@@ -0,0 +1,48 @@
+<?php
+
```
Comment (jordimorillo) [inline app/Http/Requests/SkillsRequest.php:2]:
```php
declare(strict_types=1);
```

File: app/Http/Requests/SkillsRequest.php [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
+     */
+    public function rules(): array
+    {
+        return [
+            //'skills' => 'nullable|string'
```
Comment (jordimorillo) [inline app/Http/Requests/SkillsRequest.php:37]: Remove commented code

File: app/Http/Requests/SkillsRequest.php [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+	protected function prepareForValidation()
+	{
+		$this->merge([
+			'skills' => json_encode($this->skills)
+		]);
+	}
+
+    /**
+     * Get the validation rules that apply to the request.
```
Comment (jordimorillo) [inline app/Http/Requests/SkillsRequest.php:30]: Remove unnecessary information in comments

File: app/Http/Requests/SkillsRequest.php [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+namespace App\Http\Requests;
+
+use App\Rules\DniNieRule;
+use Illuminate\Contracts\Validation\Validator;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Http\Exceptions\HttpResponseException;
+
+class SkillsRequest extends FormRequest
+{
+    /**
```
Comment (jordimorillo) [inline app/Http/Requests/SkillsRequest.php:12]: Remove comment

File: tests/Feature/Controller/Student/StudentSkillsControllerTest.php [hunk 8]
```diff
@@ -0,0 +1,84 @@
+<?php
+declare(strict_types=1);
+
+namespace Tests\Feature\Controller\Student;
+
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use Tests\TestCase;
+use Illuminate\Testing\TestResponse;
+
+//use App\Models\Student;
```
Comment (jordimorillo) [inline tests/Feature/Controller/Student/StudentSkillsControllerTest.php:10]: remove unused comments

File: tests/Feature/Controller/Student/StudentSkillsControllerTest.php [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+class StudentSkillsControllerTest extends TestCase
+{
+    use DatabaseTransactions;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+    }
+
+    private function createUserData(): array
```
Comment (jordimorillo) [inline tests/Feature/Controller/Student/StudentSkillsControllerTest.php:22]: Refactor this

File: app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+     *                 example="php:react:javascript:node:html5",
+     *             ),
+     *             @OA\Property(
+     *                 property="status",
+     *                 type="string",
+     *                 example="succes/failure",
+     *             )
+     *             )
+     *         )
+     *     ),
```
Comment (jordimorillo) [inline app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php:63]: 402

File: tests/Feature/Controller/Student/StudentSkillsControllerTest.php [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+	
+	
+	
+	static function studentSkillsControllerSuccessProvider()
+    {
+		$studentID = '4';
+		
+		$array = array(
+			array(
+				$studentID, //Sutudent ID
```
Comment (jordimorillo) [inline tests/Feature/Controller/Student/StudentSkillsControllerTest.php:58]: Remove comments

File: tests/Feature/Service/Student/StudentSkillsServiceTest.php [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+    
+    
+    static function updateSkillsByStudentIdProvider()
+    {
+
+		$studentID = '4';
+		
+		$array = array(
+			array(
+				$studentID, //Sutudent ID
```
Comment (jordimorillo) [inline tests/Feature/Service/Student/StudentSkillsServiceTest.php:61]: Remove comments

File: tests/Feature/Service/Student/StudentSkillsServiceTest.php [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+    
+    
+     /**
+     * @dataProvider updateSkillsByStudentIdProvider
+     */ 
+    public function testUpdateSkillsByStudentId(string $studentID, string $skills, bool $expectedResult): void
+    {
+		
+		
+		if($expectedResult == False){
```
Comment (jordimorillo) [inline tests/Feature/Service/Student/StudentSkillsServiceTest.php:39]: Extract exception cases.

File: tests/Feature/Service/Student/StudentSkillsServiceTest.php [hunk 14]
```diff
@@ -0,0 +1,98 @@
...
+        
+        $this->studentSkillsService = new StudentSkillsService();
+    }
+	
+    
+    
+     /**
+     * @dataProvider updateSkillsByStudentIdProvider
+     */ 
+    public function testUpdateSkillsByStudentId(string $studentID, string $skills, bool $expectedResult): void
```
Comment (jordimorillo) [inline tests/Feature/Service/Student/StudentSkillsServiceTest.php:35]: testCanUpdateStudentSkillsByStudentId

**Review by jordimorillo — CHANGES_REQUESTED** [review]
Fix this little things

File: app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php (hunk 1 above)
Comment (JustTheDude001) [reply app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php:2]: Done

File: app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php (hunk 2 above)
Comment (JustTheDude001) [reply app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php:30]: Done

File: app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+     *                 type="string",
+     *                 example="react"
+     *             ),
+     *              example={"React", "PHP", "Java", "Nodejs"}),
+     *          )
+     *      ),
+     * 
+     * 
+     *     @OA\Response(
+     *         response=204,
```
[the remark this answers is not in the document]
Comment (JustTheDude001) [reply app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php:42]: Done

File: app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+     *         description="Success. Returns the skills.",
+     * 
+     *         @OA\JsonContent(
+     *             type="array",
+     *             
+     *             @OA\Items(
+     *             type="object",
```
[the remark this answers is not in the document]
Comment (JustTheDude001) [reply app/Annotations/OpenApi/Controllers/StudentResume/StudentSkillsAnnotation.php:49]: Done

File: app/Http/Controllers/api/Student/StudentSkillsController.php (hunk 3 above)
Comment (JustTheDude001) [reply app/Http/Controllers/api/Student/StudentSkillsController.php:21]: Done

File: app/Http/Requests/SkillsRequest.php (hunk 4 above)
Comment (JustTheDude001) [reply app/Http/Requests/SkillsRequest.php:2]: Done

File: app/Http/Requests/SkillsRequest.php (hunk 6 above)
Comment (JustTheDude001) [reply app/Http/Requests/SkillsRequest.php:30]: Done

File: app/Http/Requests/SkillsRequest.php (hunk 5 above)
Comment (JustTheDude001) [reply app/Http/Requests/SkillsRequest.php:37]: Done

File: tests/Feature/Controller/Student/StudentSkillsControllerTest.php (hunk 11 above)
Comment (JustTheDude001) [reply tests/Feature/Controller/Student/StudentSkillsControllerTest.php:58]: Done

File: tests/Feature/Service/Student/StudentSkillsServiceTest.php [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+				$studentID, //Sutudent ID
+				'[]', // String
+				True //Result
+				),
+			array(
+				$studentID, //Sutudent ID
+				'["one"]', // String
+				True //Result
+				),
+			array(
```
[the remark this answers is not in the document]
Comment (JustTheDude001) [reply tests/Feature/Service/Student/StudentSkillsServiceTest.php:75]: Done

File: tests/Feature/Service/Student/StudentSkillsServiceTest.php [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+				'No User ID', //Sutudent ID
+				'["one", "two", "three"]', // String
+				False //Result
+				),
+			array(
```
[the remark this answers is not in the document]
Comment (JustTheDude001) [reply tests/Feature/Service/Student/StudentSkillsServiceTest.php:80]: Done

File: tests/Feature/Service/Student/StudentSkillsServiceTest.php (hunk 13 above)
Comment (JustTheDude001) [reply tests/Feature/Service/Student/StudentSkillsServiceTest.php]: Done

File: tests/Feature/Service/Student/StudentSkillsServiceTest.php (hunk 12 above)
Comment (JustTheDude001) [reply tests/Feature/Service/Student/StudentSkillsServiceTest.php]: Done

File: .dockerignore
Comment (JustTheDude001) [reply .dockerignore:1]: Done

File: app/Http/Requests/SkillsRequest.php (hunk 7 above)
Comment (JustTheDude001) [reply app/Http/Requests/SkillsRequest.php]: Done

File: tests/Feature/Controller/Student/StudentSkillsControllerTest.php (hunk 8 above)
Comment (JustTheDude001) [reply tests/Feature/Controller/Student/StudentSkillsControllerTest.php]: Done

File: tests/Feature/Controller/Student/StudentSkillsControllerTest.php (hunk 9 above)
Comment (JustTheDude001) [reply tests/Feature/Controller/Student/StudentSkillsControllerTest.php]: Done

File: tests/Feature/Service/Student/StudentSkillsServiceTest.php (hunk 17 above)
[the remark this answers is not in the document]
Comment (JustTheDude001) [reply tests/Feature/Service/Student/StudentSkillsServiceTest.php]: Done

[the remark this answers is not in the document]
Comment (JustTheDude001) [reply tests/Feature/Service/Student/StudentSkillsServiceTest.php]: All modifications done

**Review by jordimorillo — APPROVED** [review]

File: app/Http/Controllers/api/LoginController.php
Comment (nachoa88) [inline app/Http/Controllers/api/LoginController.php:1]: This file should be removed.

**Review by nachoa88 — APPROVED** [review]

**Comment by JustTheDude001** [comment]
Just merging

**Pull request merged**